### PR TITLE
discussion/backend

### PR DESCRIPTION
**Pull request by jackiewang5566**

### CODE REVIEW

**Comment by yijia-cc** [comment]
Per our discussion in https://github.com/yijia-cc/grouplive/pull/43, each PR should be small and focus on one issue or feature. @jackiewang5566 as the code owner of discussion, would like to check in code altogether. We discussed offline and agreed on this. But from now on, you should stick with our rule on pushing changes by feature/issue.

File: .drone.yml [hunk 1]
```diff
@@ -55,11 +55,12 @@ name: feature-discussion
 
 steps:
   - name: Run unit tests
-    image: openjdk:11.0.2-jdk-oracle
+    image: maven:3.8.1-jdk-11
```
Comment (yijia-cc) [inline .drone.yml:58]: you will also need corresponding change in Dockerfile for staging and prod deploy

Comment (jackiewang5566) [reply .drone.yml:58]: Ok, will make the change.

File: discussion/src/main/java/info/grouplive/discussion/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+//    public void configureGlobal(AuthenticationManagerBuilder authenticationManagerBuilder) throws Exception {
+//        authenticationManagerBuilder.userDetailsService(userDetailsService)
+//                            .passwordEncoder(passwordEncoder());
+//    }
+
+//    @Bean
+//    PasswordEncoder passwordEncoder() {
+//        return new BCryptPasswordEncoder();
+//    }
+}
```
Comment (yijia-cc) [inline discussion/src/main/java/info/grouplive/discussion/config/SecurityConfig.java:72]: Will these commented code be used in the future? if not, please remove them!

File: discussion/src/test/java/info/grouplive/discussion/DiscussionApplicationTests.java [hunk 3]
```diff
@@ -1,12 +1,14 @@
 package info.grouplive.discussion;
 
-import org.junit.jupiter.api.Test;
+import org.junit.Test;
+//import org.junit.jupiter.api.Test;
```
Comment (yijia-cc) [inline discussion/src/test/java/info/grouplive/discussion/DiscussionApplicationTests.java:4]:
```suggestion
```

File: discussion/src/main/resources/application.properties [hunk 4]
```diff
@@ -1 +1,17 @@
...
+spring.datasource.initialize=true
+spring.jpa.show-sql=true
+############## Mail Properties ########################
+#spring.mail.host=smtp.mailtrap.io
+#spring.mail.port=25
+#spring.mail.username=3b7e51c0c943b6
+#spring.mail.password=f3e822a129f64a
+#spring.mail.protocol=smtp
+############ JWT Properties #####################
+#jwt.expiration.time=90000
```
Comment (yijia-cc) [inline discussion/src/main/resources/application.properties:17]:
```suggestion
```

File: discussion/src/main/java/info/grouplive/discussion/service/VoteService.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.Optional;
+
+import static info.grouplive.discussion.model.VoteType.UPVOTE;
+
+@Service
+@AllArgsConstructor
+public class VoteService {
+    private final VoteRepository voteRepository;
+    private final PostRepository postRepository;
+//    private final AuthService authService;
```
Comment (yijia-cc) [inline discussion/src/main/java/info/grouplive/discussion/service/VoteService.java:24]:
```suggestion
```

File: discussion/src/main/java/info/grouplive/discussion/service/VoteService.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        voteRepository.save(mapToVote(voteDto, post));
+        postRepository.save(post);
+    }
+
+    private Vote mapToVote(VoteDto voteDto, Post post) {
+        return Vote.builder()
+                .voteType(voteDto.getVoteType())
+                .post(post)
+                .build();
+//                .user(authService.getCurrentUser())
```
Comment (yijia-cc) [inline discussion/src/main/java/info/grouplive/discussion/service/VoteService.java:51]:
```suggestion
```

File: discussion/src/main/java/info/grouplive/discussion/service/CommentService.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        commentRepository.save(comment);
+
+//        String message = mailContentBuilder.build(authService.getCurrentUser() + " posted a comment on your post." + POST_URL);
+//        sendCommentNotification(message, post.getUser());
+    }
+
+//    private void sendCommentNotification(String message, User user) {
+//        mailService.sendMail(new NotificationEmail(user.getUsername() + " Commented on your post", user.getEmail(), message));
+//    }
+
```
Comment (yijia-cc) [inline discussion/src/main/java/info/grouplive/discussion/service/CommentService.java:46]:
```suggestion
```

Comment (yijia-cc) [inline discussion/src/main/java/info/grouplive/discussion/service/CommentService.java:46]: Will these commented code be used in the future? if not, please remove them!

File: discussion/src/main/java/info/grouplive/discussion/service/CommentService.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+public class CommentService {
+    private static final String POST_URL = "";
+    private final PostRepository postRepository;
+    private final UserRepository userRepository;
+    private final CommentMapper commentMapper;
+    private final CommentRepository commentRepository;
+//    private final AuthService authService;
+//    private final MailContentBuilder mailContentBuilder;
+//    private final MailService mailService;
+
```
Comment (yijia-cc) [inline discussion/src/main/java/info/grouplive/discussion/service/CommentService.java:32]: Will these commented code be used in the future? if not, please remove them!

File: discussion/src/main/java/info/grouplive/discussion/mapper/PostMapper.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    public abstract PostResponse mapToDto(Post post);
+
+    Integer commentCount(Post post) {
+        return commentRepository.findByPost(post).size();
+    }
+
+//    String getDuration(Post post) {
+//        return TimeAgo.using(post.getCreatedDate().toEpochMilli());
+//    }
+}
```
Comment (yijia-cc) [inline discussion/src/main/java/info/grouplive/discussion/mapper/PostMapper.java:48]: Per offline discussion, will use auth serivce later, keep it for now

File: discussion/src/main/java/info/grouplive/discussion/mapper/PostMapper.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Mapper(componentModel = "spring")
+public abstract class PostMapper {
+    @Autowired
+    private CommentRepository commentRepository;
+    @Autowired
+    private VoteRepository voteRepository;
+//    @Autowired
+//    private AuthService authService;
```
Comment (yijia-cc) [inline discussion/src/main/java/info/grouplive/discussion/mapper/PostMapper.java:22]: Will use later

File: discussion/src/main/java/info/grouplive/discussion/dto/PostResponse.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+public class PostResponse {
+    private Long id;
+    private String postName;
+    private String url;
+    private String description;
+    private String userName;
+    private String subredditName;
+    private Integer voteCount;
+    private Integer commentCount;
+//    private String duration;
```
Comment (yijia-cc) [inline discussion/src/main/java/info/grouplive/discussion/dto/PostResponse.java:19]: Will these commented code be used in the future? if not, please remove them!

**Review by yijia-cc — APPROVED** [review]
Thank you very much!

**Pull request merged**